geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/RegionAdvisor.java

```diff
@@ -180,19 +180,23 @@ public void processProfilesQueuedDuringInitialization() {
               logger.trace(LogMarker.DISTRIBUTION_ADVISOR_VERBOSE,
                   "applying queued profile removal for all buckets for {}", qbp.memberId);
             }
-            for (int i = 0; i < buckets.length; i++) {
-              BucketAdvisor ba = buckets[i].getBucketAdvisor();
-              int serial = qbp.serials[i];
-              if (serial != ILLEGAL_SERIAL) {
-                ba.removeIdWithSerial(qbp.memberId, serial, qbp.destroyed);
-              }
-            } // for
+            if (qbp.serials.length == 1) {
```

Comment:
It is not clear to me why this change was made. It seems like the old code required that the the "serials" array length was >= to the "buckets.length". But now you allow "serials.length" to be 1 and in that case do not check for ILLEGAL_SERIAL. What corresponding change required this one?

Reply:
The corresponding change is in `removeIdAndBucket()` which enqueues a `QueuedBucketProfile` that has a single serial to destroy. I missed `ILLEGAL_SERIAL` checking here. The serials array is used to for `DestroyPartitionedRegionMessage`. Here we use the single element serials array for `DestroyRegionMessage`.